File: ShallowWaterSolvers/compressible/rossbywave/cacheSphi2.cpp
#include "nr.h"
#include "parameters.h"

void NR::cacheSphi2(Mat_O_DP &Sphi2, Vec_I_DP &phi)
{
	// Computes the values of the cosine phi1 basis
	// functions at each collocation point.

	for (int n=1;n<=N+1;n++)
	{
		for (int i=1;i<=N;i++)
		{
			Sphi2[n-1][i-1]=sin(2.0*n*phi[i-1]);
		}
	}
}

File: MatlabCodeAndResults/incompressible nonliear shallow water equations/Fourier Series methods/no eta in ulam zonal flow/cacheSphi2.m
function Sphi2 = cacheSphi2(N,phi)

%// Computes the values of the cosine basis
%// functions at each collocation point.

Sphi2=zeros(N+1,N);

for n=1:N+1
	for i=1:N
		Sphi2(n,i)=sin(2*n*phi(i,1));
    end
end